### PR TITLE
Improve log colors

### PR DESCRIPTION
**Pull request by felixfbecker**
It has been bothering me ever since that we log "INFO" as yellow, which I would associate with warnings.

This overhauls the log colors, and makes the code easier to understand by using the color package we already use for LSP proxy, and maps instead of arrays. 

image

### CODE REVIEW

File: cmd/gitserver/main.go [hunk 1]
```diff
@@ -33,6 +34,9 @@ var (
 )
 
 func main() {
+	// Enable colors by default but support https://no-color.org/
+	color.NoColor = env.Get("NO_COLOR", "", "Disable colored output") != ""
```
Comment (keegancsmith) [inline cmd/gitserver/main.go]: given we do this in so many files, can you add this to the `tracer.Init` function below. Amongst other things, it sorts out logging.

File: pkg/tracer/tracer.go [hunk 2]
```diff
@@ -28,21 +29,29 @@ var (
...
+	}
 	// We'd prefer these in caps, not lowercase, and don't need the 4-character alignment
-	logNames = []string{"CRIT", "ERROR", "WARN", "INFO", "DEBUG"}
+	logLabels = map[log15.Lvl]string{
+		log15.LvlCrit:  "CRITICAL",
+		log15.LvlError: "ERROR",
+		log15.LvlWarn:  "WARN",
+		log15.LvlInfo:  "INFO",
+		log15.LvlDebug: "DEBUG",
+	}
```
Comment (keegancsmith) [inline pkg/tracer/tracer.go]: nice simplification for switching to map. Technically the code before was faster which is important for logging. But I doubt that is a big deal for these logs.

File: cmd/gitserver/main.go (hunk 1 above)
Comment (felixfbecker) [reply cmd/gitserver/main.go]: I tried that but that caused panics with `env.Get must be called on package initialization`, which was very confusing. Seems like `tracer.Init()` is called after `env.Lock()` was called.

Comment (keegancsmith) [reply cmd/gitserver/main.go]: Maybe try again. Inside init should work.

Comment (felixfbecker) [reply cmd/gitserver/main.go]: The problem was `init() != Init()`

**Pull request merged**